test/e2e/kappcontroller/migrate_managed_name_test.go

```diff
@@ -0,0 +1,133 @@
+              key: value
+  template:
+    - ytt: {}
+  deploy:
+    - kapp:
+        inspect: {}`, name) + sas.ForNamespaceYAML()
+
+	// deploy a simple resource with the managed name
+	kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", oldName},
+		e2e.RunOpts{IntoNs: true, StdinReader: strings.NewReader(c)})
+
+	// kapp-controller takes over the managing of this resource as an App
+	kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name},
+		e2e.RunOpts{StdinReader: strings.NewReader(appYaml)})
+
+	_, err := kapp.RunWithOpts([]string{"inspect", "-a", name + ".app", "--raw", "--tty=false", "--filter-kind=App"}, e2e.RunOpts{AllowError: true})
+	assert.NoError(t, err, fmt.Sprintf("expected %s to exist but does not", name))
+
+	_, err = kapp.RunWithOpts([]string{"inspect", "-a", oldName, "--raw", "--tty=false", "--filter-kind=App"}, e2e.RunOpts{AllowError: true})
+	assert.Error(t, err, fmt.Sprintf("expected %s not to exist", oldName))
+}
+
+func Test_Managed_Name_Package_Repository_Migration(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	sas := e2e.ServiceAccounts{env.Namespace}
+	oldName := "simple-pkgr-ctrl"
+	name := "simple-pkgr"
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", oldName})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	c := `---
+kind: ConfigMap 
+apiVersion: v1 
+metadata:
+  name: simple-configmap-e2e-test-migration`
+
+	pkgrYaml := fmt.Sprintf(`
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageRepository
+metadata:
+  name:  %s
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  fetch:
+    inline:
+      paths:
+        packages/pkg.test.carvel.dev/pkg.test.carvel.dev.0.0.1.yml: |
+          ---
+          apiVersion: data.packaging.carvel.dev/v1alpha1
+          kind: Package
+          metadata:
+            name: pkg.test.carvel.dev.0.0.1
+          spec:
+            refName: pkg.test.carvel.dev
+            version: 0.0.1
+            template:
+              spec: {}`, name) + sas.ForNamespaceYAML()
+
+	// deploy a simple resource with the managed name
+	kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", oldName}, e2e.RunOpts{IntoNs: true, StdinReader: strings.NewReader(c)})
+
+	// kapp-controller takes over the managing of this resource as an PackageRepository
+	kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, e2e.RunOpts{StdinReader: strings.NewReader(pkgrYaml)})
+
+	_, err := kapp.RunWithOpts([]string{"inspect", "-a", name + ".pkgr", "--raw", "--tty=false", "--filter-kind=PackageRepository"}, e2e.RunOpts{AllowError: true})
+	assert.NoError(t, err, fmt.Sprintf("expected %s to exist but does not", name))
+
+	_, err = kapp.RunWithOpts([]string{"inspect", "-a", oldName, "--raw", "--tty=false", "--filter-kind=PackageRepository"}, e2e.RunOpts{AllowError: true})
+	assert.Error(t, err, fmt.Sprintf("expected %s not to exist", oldName))
```

Comment:
would be nice to have an assertion that error is indicating "thing" is not found.